universe/syncer.go

```diff
@@ -243,22 +252,42 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot BaseRoot,
 					"invalid", spew.Sdump(key))
 			}
 
-			// TODO(roasbeef): inclusion w/ root here, also that
-			// it's the expected asset ID
-
-			fetchedLeaves <- &IssuanceItem{
-				ID:   uniID,
-				Key:  key,
-				Leaf: leafProof.Leaf,
+			// If this is an issuance proof, then we can send
+			// things directly to the batch insertion goroutine.
+			// Otherwise, we'll another step to the pipeline below
+			// for sorting.
+			if isIssuanceTree {
+				fetchedLeaves <- &IssuanceItem{
+					ID:   uniID,
+					Key:  key,
+					Leaf: leafProof.Leaf,
+				}
+			} else {
+				transferLeafProofs <- &IssuanceItem{
+					ID:   uniID,
+					Key:  key,
+					Leaf: leafProof.Leaf,
+				}
 			}
 
 			return nil
-		},
-	)
+		})
 	if err != nil {
 		return err
 	}
 
+	// If this is a tranfer tree, then we'll collect all the items as we
+	// need to sort them to ensure we can validate them in dep order.
+	if !isIssuanceTree {
+		transferLeaves := fn.Collect(transferLeafProofs)
+		sort.Slice(transferLeaves, func(i, j int) bool {
```

Comment:
I think this'll work for now. Though I wonder: Will we ever want to make it possible to have subsequent transfers within the same block? Assuming an interactive scheme where participants would accept proofs as being valid before they have the block header populated (e.g. in a pocket universe construction)?

Reply:
Yeah I think so, then we would use something like an `scid` here, that takes into account: tx index, and output index as well.

Reply:
Will this sort be run again if there's a block reorg?